### PR TITLE
Add fx3687 [v98] Wallpaper networking and deeplinks

### PR DESCRIPTION
**Pull request by adudenamedruby**

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -1240,7 +1252,7 @@ class BrowserViewController: UIViewController {
 
         if currentViewController != self {
             _ = self.navigationController?.popViewController(animated: true)
-        } else if let urlBar = urlBar, urlBar.inOverlayMode {
```
Comment (lmarceau) [inline Client/Frontend/Browser/BrowserViewController.swift:1243]: We need to revert this change - this was a fix on main

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -1474,7 +1486,7 @@ extension BrowserViewController: TabDelegate {
 
         let sessionRestoreHelper = SessionRestoreHelper(tab: tab)
         sessionRestoreHelper.delegate = self
-        tab.addContentScriptToPage(sessionRestoreHelper, name: SessionRestoreHelper.name())
+        tab.addContentScript(sessionRestoreHelper, name: SessionRestoreHelper.name())
```
Comment (lmarceau) [inline Client/Frontend/Browser/BrowserViewController.swift:1489]: Same here, need to revert

File: Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            self.browserLog.debug("Wallpaper - error writing file to disk: \(error)")
+            throw error
+        }
+    }
+
+    // MARK: - Wallpaper retrieval
+    func getCurrentWallpaperObject() -> Wallpaper? {
+        if let savedWallpaper = userDefaults.object(forKey: PrefsKeys.WallpaperManagerCurrentWallpaperObject) as? Data {
+            let decoder = JSONDecoder()
+            if let wallpaper = try? decoder.decode(Wallpaper.self, from: savedWallpaper) {
```
Comment (lmarceau) [inline Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift:104]: return try? decoder.decode(Wallpaper.self, from: savedWallpaper)

File: Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                return wallpaper
+            }
+        }
+
+        return nil
+    }
+
+    func getCurrentWallaperImage() -> UIImage? {
+        let key = UIDevice.current.orientation.isLandscape ? PrefsKeys.WallpaperManagerCurrentWallpaperImageLandscape : PrefsKeys.WallpaperManagerCurrentWallpaperImage
+        guard let image = getImageResource(for: key) else { return nil }
```
Comment (lmarceau) [inline Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift:114]: return getImageResource(for: key)

File: Client/Frontend/Home/Wallpapers/WallpaperResourceManager.swift [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    init(folder: String, portrait: String, landscape: String) {
+        self.folder = folder
+        self.portrait = portrait
+        self.landscape = landscape
+    }
+}
+
+class WallpaperResourceManager {
+    
+    init() { }
```
Comment (lmarceau) [inline Client/Frontend/Home/Wallpapers/WallpaperResourceManager.swift:39]: necessary?

File: Client.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -687,7 +691,7 @@
 		D3FA777B1A43B2990010CD32 /* SearchTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = D3FA777A1A43B2990010CD32 /* SearchTests.swift */; };
 		D3FEC38D1AC4B42F00494F45 /* AutocompleteTextField.swift in Sources */ = {isa = PBXBuildFile; fileRef = D3FEC38C1AC4B42F00494F45 /* AutocompleteTextField.swift */; };
 		D433852C27ABC8150069DD33 /* MappaMundi in Frameworks */ = {isa = PBXBuildFile; productRef = D433852B27ABC8150069DD33 /* MappaMundi */; };
-		D433852D27ABC8250069DD33 /* RUSTPlaces.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 43BE57EA278BA4ED00491291 /* RUSTPlaces.framework */; };
+		D433852D27ABC8250069DD33 /* (null) in Frameworks */ = {isa = PBXBuildFile; };
```
[the remark this answers is not in the document]
Comment (electricRGB) [reply Client.xcodeproj/project.pbxproj:694]: Checked with Nish. RUSTPlaces being removed is fine. Null is xcode being weird. :/

[the remark this answers is not in the document]
Comment (lmarceau) [reply Client.xcodeproj/project.pbxproj:694]: Haha well then, that'll be it!

**Review by lmarceau — APPROVED** [review]

**Comment by electricRGB** [comment]
[Tests passed.](https://app.bitrise.io/build/dc7e0b92-547b-432b-97cc-e818d34ce27d#?tab=log)
